### ISSUE TITLE
why Rithvik Jana and Chande repeatedly grading the mission ? How to model the mission plan review process

### ISSUE THREAD

**Issue by shirishag75** [mars-sim/mars-sim#265]
See the log, that will tell all. 
[mars_simu_0.log](https://github.com/mars-sim/mars-sim/files/4006872/mars_simu_0.log)

**Comment by Urwumpe** [comment]
Pay attention to the part that says 

> 70.0% review completed

**Comment by shirishag75** [comment]
@Urwumpe where do you get that. I get 100% review completed

See at 00-Adir-01:225.947 or are we seeing two different things. 

Later I saw the same thing with the commander's mission i.e. shirish agarwal also reviewing and re-reviewing his own mining plan and going ahead with it. I thought the idea was that other people in the group are/were going to grade your mission plan which would bring in an element of randomness as well as unique perspectives as it would happen to any plan or discussion over anything (just like in real life.) I also have that one saved up if that would make it more interesting.

**Comment by Urwumpe** [comment]
> 
> 
> @Urwumpe where do you get that. I get 100% review completed

In your log, reading the blocks of lines chronologically. 

See here, filtered for the respective lines:
```

00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 27.8 (10.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 58.6 (20.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 86.4 (30.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 112.2 (40.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 140.0 (50.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 171.8 (60.0% review completed)
00-Adir-01:225.947 (Info) MissionManager : Rithvik Jana's Regolith Prospecting mission plan - current score : 196.6 (70.0% review completed)
...
```


What I just find strange there is that the reviews take no time and are done in sequence. And yes, having a full "self-review" also looks strange. I maybe use a different way to calculate the progress, so that not every review is a ten percent step.

**Comment by shirishag75** [comment]
I find that whole thing strange. I also saw that as I said later as well, for the same thing happened for similar mission by shirish agarwal (the commander) too. 

Btw it seems you didn't see this - 

```
00-Adir-01:225.947 (Info) ReviewMissionPlan : [New Mumbai] Rithvik Chande approved Rithvik Jana's Regolith Prospecting mission plan. Total Score: 280.0 (Min: 230.0).
00-Adir-01:264.557 (Info) Mission : [New Mumbai] Engineering Specialist Rithvik Jana was getting ready to embark on Regolith Prospecting
```

**Comment by mokun** [comment]
> shirish agarwal also reviewing and re-reviewing his own mining plan

> look at 00-Adir-02:605.157 where shirish approves his own mission and the steps preceeding that.

So this is a design question: should we allow commander to grade his own plan ?

**Comment by mokun** [comment]
> 00-Adir-01:225.947 (Info) ReviewMissionPlan : [New Mumbai] Rithvik Chande approved Rithvik Jana's Regolith Prospecting mission plan. Total Score: 280.0 (Min: 230.0).
00-Adir-01:264.557 (Info) Mission : [New Mumbai] Engineering Specialist Rithvik Jana was getting ready to embark on Regolith Prospecting

That tells me the mission plan is 100% fully graded. 

At this point, if the total score is larger than the minimum score, the mission plan is approved.

The minimum score gets updated all the time and is the average of the previous two passing/non-passing scores.

**Comment by mokun** [comment]
> What I just find strange there is that the reviews take no time and are done in sequence. And yes, having a full "self-review" also looks strange. I maybe use a different way to calculate the progress, so that not every review is a ten percent step.

@Urwumpe , I'll fix this and make the review process spread out over a period of time.

**Comment by mokun** [comment]
Q: If you are a commander, how would you decide how to grade a mission plan ?

For settlement with a large number of people, we can have some selected folks grading it. 

But for small settlement, we have less folks available. Say for a 4-person settlement, due to shift rotation, 2 of them can be asleep and 2 are working. 

So in this case, may be we want *everybody* to look at a mission plan since it would impact everyone.

But for a 8-person, do we need to have everyone look at a mission plan ?

**Comment by shirishag75** [comment]
> ** EDITED **
> 
> Q1: should we allow the same person to grade the same mission plan ?

Sure, but one person, one vote that kind of thing. 

> Q2: what options should be available for a commander to decide how to grade a mission plan ?

Maybe a bit more weightage than a normal crew member but not by much, at least not in the beginning. Later, over years with experience and trust of others he could do more but at the beginning  only little more. How do we compute experience and gaining trust is something I leave to you. 

> For settlement with a large number of people, we can have some selected folks grading it.

Taking from Star Trek perspective, all the away teams had the commander/captain, the vice-captain or first officer, the doctor (the health perspective) , the security officer (who is responsible for the security of the ship) and the science officer  (who oversees the various sciences which help both in survival and enrichment of the colony/ship) . At least these 4-5 voices for sure.  

> But for small settlement, we have less folks available. Say for a 4-person settlement, due to shift rotation, 2 of them can be asleep and 2 are working.
> 
> So in this case, may be we want _everybody_ to look at a mission plan since it would impact everyone.
>

On the money. 
 
> But for a 8-person, do we need to have everyone look at a mission plan ?

I think I shared it above, although another thing to point out is I dunno how well rescue missions are thought of, there should be procedures for it. As should be procedures for farming and having more farms etc. Although right now, when we do regolith mining, what are we doing the regolith mining and processing for, that is also not clear to me. 

The only mission objective is to do regolith mining and processing but the processing is being done for what is not known or cleared up. What the processed regolith would be used for needs to be known before. At some point we would need to also tutorialize the objectives so we know what is needed, but that perhaps is sometime away.

**Comment by Urwumpe** [comment]
Important though, there should be always only one person that has the responsibility for the approval. Can we maybe add some historic record for any general approval process or command decision in the simulation? Would make it easier to later find out how a mission got approved or rejected.

**Comment by shirishag75** [comment]
> Important though, there should be always only one person that has the responsibility for the approval. Can we maybe add some historic record for any general approval process or command decision in the simulation? Would make it easier to later find out how a mission got approved or rejected.

Did you see The Martian ?  When the storm is about to hit the mars, all chime in. Even in ISS  or even the Mars rovers I'm sure tens of people would argue back and forth as to what is the best way to do things rather than one just person taking all the decision. At the very least s/he would listen to all the viewpoints and then make an **informed** decision. Now because we do not have a way to make the commander informed as shared above, I think this is the next best way. 

@mokun let us know what you think as well.

**Comment by Urwumpe** [comment]
> Did you see The Martian ? When the storm is about to hit the mars, all chime in. Even in ISS or even the Mars rovers I'm sure tens of people would argue back and forth as to what is the best way to do things rather than one just person taking all the decision. At the very least s/he would listen to all the viewpoints and then make an **informed** decision. Now because we do not have a way to make the commander informed as shared above, I think this is the next best way.

Yes - and one commander had the final word there.  Informed decision does not mean, that all decide. Informed decision means that all experts for a specific question within a complex decision are consulted and in the ideal world, the single accountable decision maker decides based on the answers he/she got from the experts - and the recommendations by the various experts might even be contradicting and the commander would have to find a compromise.

**Comment by shirishag75** [comment]
@Urwumpe and @mokun in such a scenario where the commander takes the last call on everything then why do we even need the relationship score. Let it be just be a lottery thing, chance thing or Russian Roulette. Because the captain/commander doesnt need to explain any of his explanations to the group or anybody else. It would make the whole thing whole lot simpler if you wanna go that way. Then you don't have any risk-assessment methodology as it's just what seems good or bad for the captain and he has no guide AFAIK to help her/him make a decision one way or the other atm.

**Comment by mokun** [comment]
> why do we even need the relationship score

In real world, relationship impacts everything we do :D

**Comment by mokun** [comment]
> When the storm is about to hit the mars, all chime in. Even in ISS or even the Mars rovers I'm sure tens of people would argue back and forth as to what is the best way to do things rather than one just person taking all the decision. 

We could create an abstract model of a group debate.

We may create different style of leadership and how each style impact the decision making process of for instance what to do when a disaster strike.

**Comment by mokun** [comment]
> Important though, there should be always only one person that has the responsibility for the approval. 

I can see your point. It's a lot more efficient if a commander has **the last word** for approving or disapproving a mission, rather than going through a debate process. 

Or: we may simply let people in that settlement vote either *FOR* or *AGAINST* a mission.

> Can we maybe add some historic record for any general approval process or command decision in the simulation? Would make it easier to later find out how a mission got approved or rejected.

Yes. Right now, unapproved mission plan are being deleted from the 'Mission' tab. 

I supposed we can save each proposal but storage will become an issue in long run.

**Comment by shirishag75** [comment]
> > When the storm is about to hit the mars, all chime in. Even in ISS or even the Mars rovers I'm sure tens of people would argue back and forth as to what is the best way to do things rather than one just person taking all the decision.
> 
> We could create an abstract model of a group debate.
> 
> Also we could model different style of leadership and how each style uniquely impact the decision-making process of for instance what to do when a disaster strike.

The question of the disaster strike was an extreme take or the ones in Martian but the reality is unless and until you do not have a debate you cannot have a nuanced view and have a naunced decision. What @Urwumpe fails to recognize that from what I have seen  in Star Trek, The Martian (popular media) and elsewhere the captain had to explain her/his decisions to superiors. In fact, in Star Trek when the captain goes on 'Away' Missions s/he always used her personal recorder to record the decisions and rationale she used while making decisions. Her decisions and the steps and nuanced views s/he takes as a captain and the reasoning are often what saves her in court of law either on Earth or in alien enemy camps where she along with the help of her first officer she is able to show good faith. 

I am guessing you now know why I asked for this. It also becomes a matter or record-keeping and using both the successes and failures to be used in future environments while training astronauts for off-world adventures as well as seeing how their behavior is, sane, insane, empathatic or otherwise. Even 5 years on Mars could take a psychological toll on people as we have seen in limited experiments on earth as to what isolation can do to people, especially Russian and American experiments. 

As far as storage issues are concerned, I think that only will be fixed more or less when we have serialized .json . Once we have simplified that, it perhaps will be far more easier to compress and de-compress the save file and figure out how things are from the save-file itself.

**Comment by mokun** [comment]
> In fact, in Star Trek when the captain goes on 'Away' Missions s/he always used her personal recorder to record the decisions and rationale she used while making decisions. 

Sure. In real world, the commander or the captain is responsible for getting the mission done as well as being accountable for the welfare of his crew.

My question is : how do we model this in a computer simulation ?

> Her decisions and the steps and nuanced views s/he takes as a captain and the reasoning are often what saves her in court of law either on Earth or in alien enemy camps where she along with the help of her first officer she is able to show good faith.

Yay this makes things a lot of interesting in the plot.

The superior of the commander/captain usually have desk jobs. The type of risks they are confronted with is : can I trust this captain with a spaceship ? Do I have the political capital to play it out. They are politician by the virtue of what they do.

But in case of mars-sim, the commander of a settlement should report directly to the **Mission Control** folks of the sponsoring space agency.

**Comment by shirishag75** [comment]
> > In fact, in Star Trek when the captain goes on 'Away' Missions s/he always used her personal recorder to record the decisions and rationale she used while making decisions.
> 
> Sure. In real world, the commander or the captain is responsible for getting the mission done as well as being accountable for the welfare of his crew.
> 
> My question is : how do we model this in a computer simulation ?
> 
> > Her decisions and the steps and nuanced views s/he takes as a captain and the reasoning are often what saves her in court of law either on Earth or in alien enemy camps where she along with the help of her first officer she is able to show good faith.
> 
> Yay this makes things a lot of interesting in the plot.
> 
> The superior of the commander/captain usually have desk jobs. The type of risks they are confronted with is : can I trust this captain with a spaceship ? Do I have the political capital to play it out. They are politician by the virtue of what they do.
> 
> But in case of mars-sim, the commander of a settlement should report directly to the **Mission Control** folks of the sponsoring space agency.

Usually but that is when you have a large army having thousands of people under your command where you have to spend more time in strategizing, while if you are the captain with boots on ground you do not have that luxury. In the above scenario you are talking about a four-five star general if I were to use military terms. 

Coming from the Indian context there have been numerous examples of Indian officers, captains leading from the front. I could give quite a few examples of people who made the ultimate sacrifice for the nation. 

https://zeenews.india.com/india/indian-army-woman-officer-captain-kalpana-kundu-breaks-barriers-undertakes-high-altitude-patrol-in-himalayas-2213607.html

See also https://www.quora.com/Why-are-senior-officers-of-the-Indian-army-dying-on-the-frontlines-during-terrorist-attacks

https://www.ssbcrack.com/2016/01/5-commanding-officers-who-led-from-the-front-and-became-heroes.html

There probably are many more. My point is the team on Mars should know and have explicit and implicit trust in their captain that he has got their back and the same is with him/her . 

Of the few examples I shared,  I think I should share a bit of the background of the first. For e.g. women officers in the three forces were always given short service commission which they felt was the wrong thing as the guys used to have full service and enjoy pensions while they do not. They went to the highest court of the country and got quite a few decisions in their favor. The Defence Minister who is also a political animal at the time had lot of pressure from women to have more participation of women in the armed forces and grudginly the three forces had to pave ways for both full service comissions as well as treating them as equals. While as can be seen there is lot of work yet to be done for bolstering the ranks of the women officers to be equal to men, a beginning has been made in the last few years.

**Comment by mokun** [comment]
@shirishag75 
> There probably are many more. My point is the team on Mars should know and have explicit and implicit trust in their captain that he has got their back and the same is with him/her .

This is in the arena of social simulation. 

We can google research journal out there to see the computational or programming approaches of modeling inter-personal relationship and command relationship. 

We can add `trustworthiness` or `integrity` as a natural attribute. 

@Urwumpe 

> Yes - and one commander had the final word there. Informed decision does not mean, that all decide.

I've thought of creating our own syntax for interaction between settlers using some sort of Agent Simulation and Model approaches.

**Comment by mokun** [comment]
@shirishag75  @Urwumpe ,

I know we've discussed many different possible improvement we can have in this thread. 

For now, I can only do one thing at a time :D

The [commit](https://github.com/mars-sim/mars-sim/commits/master) I just made was a slight refinement of calculating the site value of a possible mission.

In the past, we only calculate the site score if it's an ice collection mission.

Now, we include the following 4 types of missions, namely, the ice, regolith, exploration and mining mission.

Let's close this thread and open any new ones you want to work on.
